examples/mnist.py

```diff
@@ -1,10 +1,20 @@
 from __future__ import print_function
 
-import cPickle as pickle
 import gzip
 import itertools
-import urllib
+import pickle
 import os
+import sys
+
+PY2 = sys.version_info[0] == 2
+
+if PY2:
+    from urllib import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f)
+else:
+    from urllib.request import urlretrieve
+    pickle_load = lambda f, encoding: pickle.load(f, encoding=encoding)
+
```

Comment:
> I wonder what's going on here.

Must be because of this pickle's content; it's mostly numpy arrays that probably won't make much of a difference between `pickle` and `cPickle`. It's still a bit of a mystery, but now that `pickle` is faster for this example, it's more environmentally friendly and I'm fine with keeping it!